concepts/modularization/modularization_overview.md

```diff
@@ -0,0 +1,60 @@
+
+Module Manager retrieves the image layers from the specified image registry and processes the manifest resources by calling a rendering framework like Helm or Kustomize.
+
+Finally, it deploys this resource along with a custom resource to track state changes. It ensures consistency both with a time-based reconciliation and by watching state changes of the deployed custom resource in Kyma runtime.
+
+## 3. Runtime Watcher
+
+> **NOTE:** Runtime Watcher is a necessary component only when running with a control plane that manages one or more other runtimes. For local (single-cluster) setups, the Runtime Watcher is not needed.
+
+Runtime Watcher monitors the relevant module resources of the user’s Kyma runtime for configured changes, specified by the operator in the control plane, such as:
+
+* Expected changes, like a Kyma user providing a configuration update to the module, which must lead to an update of related resources in the Kyma runtime.
+* Unexpected changes, like an accidental creation, deletion, or modification, which must be reverted to recover the Kyma runtime to a healthy state.
+
+For all such detected changes, Runtime Watcher triggers a reconciliation of the user’s Kyma runtime by propagating the desired changes to Module Manager and Lifecycle Manager.
+
+## 4. Lifecycle Manager
+
+Lifecycle Manager is responsible for orchestrating Kyma module operators to process their respective resources on the Kyma runtime. It also aggregates all module states into one easy-to-monitor resource, the `Kyma CR`, with the following strategies:
+
+1. Generating the required Kyma module custom resources through a manifest, which starts the Kyma module processing in Module Manager.
+
+2. Reconciling a Kyma custom resource for each Kyma runtime, indicating the consolidated status of all modules configured for that Kyma cluster.
+
+Lifecycle Manager is also responsible for propagating updates from a module into the respective manifest through the ModuleTemplate. Whenever modules are updated and a new release is triggered, Lifecycle Manager propagates the correct upgrade information to the manifest, which in turn leads to updates in the runtime cluster.
+
+## 5. ModuleTemplate
+
+The ModuleTemplate is a custom resource that is used by Lifecycle Manager as a scaffolding when initializing new modules. It is also used as a base output of the module bundling process inside the CLI.
+
+Whenever a new module must be installable or registered in the ecosystem, users only have to apply a new ModuleTemplate with the necessary information for generating the module. Then, it is available for consumption in every runtime instance registered in the control plane.
```

Comment:
```suggestion
Whenever a new module must be installed or registered in the ecosystem, you only have to apply a new ModuleTemplate with the necessary information to generate the module. Then, it is available for consumption in every runtime instance registered in the control plane.
```

Reply:
I'm not quite sure whether it must be installed, or whether it just must be possible to install it.
@tobiscr @jakobmoellersap @adityabhatia can you reply?

Reply:
possible to install. ModuleTemplate only enables installation, it doesnt actually perform the installation

Reply:
Got it. I updated the comment.